datasetinsights/datasets/base.py

```diff
@@ -1,43 +1,75 @@
-from abc import ABCMeta, abstractmethod
+from abc import abstractmethod
 
 
-class Dataset(metaclass=ABCMeta):
-    """Abstract base class for datasets.
-    """
+class DownloaderRegistry:
+    """ The Registry class for datasets"""
 
-    @staticmethod
-    def create(name, **kwargs):
-        """Create a new instance of the dataset subclass
+    dataset_registry = {}
+    download_registry = {}
+    """ Internal registry for available Datasets """
 
+    @classmethod
+    def register(cls, name: str, downloaders: list):
```

Comment:
Why do we need a list of downloaders?

Reply:
This is because a dataset can be downloaded in more then one way. example gs:// http:// or usim://